include/commissioner/commissioner.hpp

```diff
@@ -194,90 +294,16 @@ class Commissioner
-     *
-     * @param[in]  aJoinerInfo         A joiner info indexing the commissioning joiner.
-     * @param[in]  aVendorName         A human-readable product vendor name string in utf-8 format.
-     * @param[in]  aVendorModel        A human-readable product model string.
-     * @param[in]  aVendorSwVersion    A utf-8 string that specifies the product software version.
-     * @param[in]  aVendorStackVersion A vendor stack version of fixed length (5 bytes). See section
-     *                                 8.10.3.6 of Thread spec for detail.
-     * @param[in]  aProvisioningUrl    A URL encoded as a utf-8 string provided by the Joiner
-     *                                 to communicate to the user which Commissioning application
-     *                                 is best suited to properly provision it to the appropriate
-     *                                 service. Empty if the joiner doesn't provide it.
-     * @param[in]  aVendorData         A product vendor-defined data structure to guide
-     *                                 vendor-specific provisioning. Empty if the joiner doesn't provide it.
-     *
-     * @return  A boolean indicates whether the joiner is accepted.
-     *
-     * @note This will be called when A well-formed JOIN_FIN.req has been received.
-     *
-     */
-    using CommissioningHandler = std::function<bool(const JoinerInfo & aJoinerInfo,
-                                                    const std::string &aVendorName,
-                                                    const std::string &aVendorModel,
-                                                    const std::string &aVendorSwVersion,
-                                                    const ByteArray &  aVendorStackVersion,
-                                                    const std::string &aProvisioningUrl,
-                                                    const ByteArray &  aVendorData)>;
-
     /**
      * @brief Create an instance of the commissioner.
      *
-     * @param[in] aConfig     A commissioner configuration.
-     * @param[in] aEventBase  A libevent event_base object. nullable.
+     * @param[in] aHandler  A handler of commissioner events.
+     * @param[in] aConfig   A commissioner configuration.
      *
      * @return std::shared_ptr<Commissioner>  nullptr is returned if failed.
      *
-     * @note If @p aEventBase is not provided, commissioner will create it by itself and
-     *       run the event loop in separate background thread. In this case, all callback
-     *       function will be called in separate thread and synchronization is needed;
-     *       Otherwise, it is the user to run the event loop and no synchronization is
-     *       needed in callback functions.
      */
-    static std::shared_ptr<Commissioner> Create(const Config &aConfig, struct event_base *aEventBase);
+    static std::shared_ptr<Commissioner> Create(CommissionerHandler &aHandler, const Config &aConfig);
```

Comment:
```suggestion
    static std::shared_ptr<Commissioner> Create(const CommissionerHandler &aHandler, const Config &aConfig);
```

Reply:
We cannot make it const, because those handlers are not `const` methods.